components/nexus-bootstrap/src/main/java/org/sonatype/nexus/bootstrap/jetty/ConnectorManager.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.bootstrap.jetty;
+
+import java.util.IdentityHashMap;
+import java.util.Map;
+
+import org.eclipse.jetty.http.HttpScheme;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+/**
+ * Jetty's internal connector manager to manage custom {@link ConnectorConfiguration}s.
+ * <p/>
+ * Note: This class relies on "id" properties in Jetty XML configuration! Hence, the existing beans in Jetty XML
+ * should NOT be renamed, while adding new (with unique, non-clashing) IDs is okay thing to do.
+ */
+public class ConnectorManager
+{
+  private static final String HTTP_CONFIG_ID = "httpConfig";
+
+  private static final String HTTP_CONNECTOR_ID = "httpConnector";
+
+  private static final String SSL_CONTEXT_FACTORY_ID = "sslContextFactory";
+
+  private static final String HTTPS_CONFIG_ID = "httpsConfig";
+
+  private static final String HTTPS_CONNECTOR_ID = "httpsConnector";
+
+  private final Server server;
+
+  private final Map<String, Object> namedBeans;
+
+  private final IdentityHashMap<ConnectorConfiguration, ServerConnector> managedConnectors;
+
+  public ConnectorManager(final Server server,
+                          final Map<String, Object> namedBeans)
+  {
+    this.server = server;
+    this.namedBeans = namedBeans;
+    this.managedConnectors = new IdentityHashMap<>();
+  }
+
+  /**
+   * Adds and starts connector to Jetty based on passed in configuration.
+   */
+  public ServerConnector addConnector(final ConnectorConfiguration connectorConfiguration)
+  {
+    verifyConfiguration(connectorConfiguration.getScheme());
+
+    final HttpConfiguration httpConfiguration =
+        connectorConfiguration.customize(defaultHttpConfiguration(connectorConfiguration.getScheme()));
```

Comment:
four calls in this block to connectorConfiguration.getScheme(), please extract to a var for readability